src/main.c

```diff
@@ -992,39 +1141,34 @@ static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_
             use_picotls = 0;
         }
     }
-    if (ocsp_update_interval_node != NULL) {
-        if (h2o_configurator_scanf(cmd, *ocsp_update_interval_node, "%" SCNu64, &ocsp_update_interval) != 0)
+
+    /* setup OCSP stapling context as `ocsp_stapling`, or set to NULL if disabled */
+    struct listener_ssl_ocsp_stapling_t *ocsp_stapling = h2o_mem_alloc(sizeof(*ocsp_stapling));
+    *ocsp_stapling = (struct listener_ssl_ocsp_stapling_t){
+        .interval = 4 * 60 * 60, /* default update interval of 4 hours */
+        .max_failures = 3,       /* by default, permit 3 consecutive failures before temporary disabling OCSP stapling */
+        .cmd = "share/h2o/fetch-ocsp-response",
+    };
+    if (ocsp_update_cmd != NULL)
+        ocsp_stapling->cmd = h2o_strdup(NULL, (*ocsp_update_cmd)->data.scalar, SIZE_MAX).base;
+    if (ocsp_max_failures_node != NULL) {
+        if (h2o_configurator_scanf(cmd, *ocsp_max_failures_node, "%u", &ocsp_stapling->max_failures) != 0)
             goto Error;
     }
-    if (ocsp_max_failures_node != NULL) {
-        if (h2o_configurator_scanf(cmd, *ocsp_max_failures_node, "%u", &ocsp_max_failures) != 0)
+    if (ocsp_update_interval_node != NULL) {
+        if (h2o_configurator_scanf(cmd, *ocsp_update_interval_node, "%" SCNu64, &ocsp_stapling->interval) != 0)
             goto Error;
+        if (ocsp_stapling->interval == 0)
+            ocsp_stapling = NULL;
```

Comment:
Looks like `ocsp_stapling` leaks here.
```suggestion
        if (ocsp_stapling->interval == 0) {
            free(ocsp_stapling);
            ocsp_stapling = NULL;
}
```

Reply:
Right. Due to src/main.c being part of a daemon rather than a library, we do not need to release resources before we exit. That's the pattern we have, we do not release resources when we jump to `Error`.

Reply:
Okay makes sense. I'll close the discussion then.